### DISCUSSION TITLE
Method to get last IReplicateData on server but also replaying clients

### DISCUSSION THREAD

**Discussion by Baerikus** [FirstGearGames/FishNet#456] [Approved Changes or Features]
It would be awesome to have a method to get the last IReplicateData on server but also replaying clients to use it inside the replicate functions. 

For some features it's required to check last inputs and if they are already internally there and could be exposed like you told on the discord https://discord.com/channels/424284635074134018/1104850498290462821/1148576097332625451 it would eliminate unnecessary solutions like ring buffering the inputs on the client, to be able to track back on replay, or adding the last input to the reconcile data, what would increase bandwidth usage quite a bit. 

If this applies only to the last input packet of not already replayed data or is generally possible to tick specific until a certain point in the past like "IReplicateData .GetDataForTick(unit tick)" or similar, is for you to decide. 
But I think a "IReplicateData .GetLastData()" or similar, while replaying replicates would serve most scenarios.

Thanks for considering,

Baerikus

**Comment by FirstGearGames** [comment]
Bump. Still want to get this in, but backlogged.

**Comment by FirstGearGames** [comment]
This is possible and had been partially implemented. The big hold is that implementing this requires more cecil codegen, which is always a nightmare. There's also a cost to storing the last created data, checking if it exists, and clearing it.

After thinking it over quite a bit, since the perf would affect all users, and not all users would want this, I think it might be best to continue developers to manage this manually.
